pkg/sessions/dynamodb/client.go

```diff
@@ -0,0 +1,74 @@
+package dynamodb
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	dynamo "github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"time"
+)
+
+type Client struct {
+	*dynamo.DynamoDB
+}
+
+var TableName = "Session"
+
+type dynamoItem struct {
+	SessionKey string
+	Value      string
+}
+
+func newClient(c *dynamo.DynamoDB) *Client {
+	return &Client{DynamoDB: c}
+}
+
+func (c *Client) Get(_ context.Context, key string) ([]byte, error) {
+	result, err := c.GetItem(&dynamo.GetItemInput{
+		TableName: aws.String(TableName),
+		Key: map[string]*dynamo.AttributeValue{
+			"SessionKey": {
+				S: aws.String(key),
+			},
+		},
+	})
+	item := dynamoItem{}
+
+	err = dynamodbattribute.UnmarshalMap(result.Item, &item)
+	if err != nil {
+		// TODO: Return default val?
+		return nil, fmt.Errorf("failed unmarshalling record: %v", err)
+	}
+
+	val, err := b64.StdEncoding.DecodeString(item.Value)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode dynamodb session: %v", err)
+	}
+
+	return val, nil
+}
+
+func (c *Client) Set(ctx context.Context, key string, value []byte, expiration time.Duration) error {
+	i := dynamoItem{
+		SessionKey: key,
+		Value:      b64.StdEncoding.EncodeToString(value),
```

Comment:
Do we need to take our `[]byte` and base64 to get a string to use in a `String` type column in the DB? It supports a `Byte` type natively, right? Can we just put the AES-GCM encrypted `[]byte` that will be passes into this method into DynamoDB directly without the overhead of encoding/decoding?

Reply:
@NickMeves So it does (have native `byte` support), reverted.